caddy/caddymain/run.go

```diff
@@ -409,13 +416,84 @@ func initTelemetry() error {
 	return nil
 }
 
+// LoadEnvFromFile loads additional envs if file provided and exists
+// Envs in file should be in KEY=VALUE format
+func LoadEnvFromFile(envFile string) error {
+	if envFile == "" {
+		return nil
+	}
+
+	file, err := os.Open(envFile)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	envMap, err := ParseEnvFile(file)
+	if err != nil {
+		return err
+	}
+
+	for k, v := range envMap {
+		if err := os.Setenv(k, v); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// ParseEnvFile implements parse logic for environment files
+func ParseEnvFile(envInput io.Reader) (map[string]string, error) {
+	envMap := make(map[string]string)
+
+	scanner := bufio.NewScanner(envInput)
+	var line string
+	lineNumber := 0
+
+	for scanner.Scan() {
+		line = scanner.Text()
+		lineNumber++
+
+		// skip lines starting with comment
+		if strings.HasPrefix(line, "#") {
+			continue
+		}
+
+		// skip empty line
+		if len(line) == 0 {
+			continue
+		}
+
+		fields := strings.SplitN(line, "=", 2)
+		if len(fields) != 2 {
+			return nil, fmt.Errorf("Can't parse line %d; line should be in KEY=VALUE format", lineNumber)
+		}
+
+		key := strings.TrimSpace(fields[0])
```

Comment:
considering a key cannot have a trailing whitespace anyway, this is another case that trimming the line of whitespace would've handled.

Reply:
Actually it can if you split line by '=' space and someone puts it like this:
```
A = B
```

then actual key would be "A " and not "A". Its pretty hard to understand and debug from the user point of view. I prefer to leave this trim space and add another one to whole line.

Reply:
`A = B` is not a valid env syntax.

Reply:
Since Docker is your reference point, here is what Docker gives.
![screenshot from 2018-05-24 11-40-30](https://user-images.githubusercontent.com/240448/40480806-78d73a80-5f47-11e8-9276-779158c0a519.png)

Reply:
Yep, you are right:) I didn't think that i can return error on this case:)